### PR TITLE
Add support for text-decoration-line

### PR DESCRIPTION
**Pull request by jacobp100**

### CODE REVIEW

File: src/transforms/textDecorationLine.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  let didParseFirst = false
+  while (tokenStream.hasTokens()) {
+    if (didParseFirst) tokenStream.expect(SPACE)
+
+    lines.push(tokenStream.expect(LINE))
+
+    didParseFirst = true
+  }
+
+  lines.sort().reverse()
```
Comment (kristerkari) [inline src/transforms/textDecorationLine.js]: oh this is clever trick, though it would not work if there were other values supported :)

**Comment by kristerkari** [comment]
Would this code throw if you use?

```css
text-decoration-line: none underline;
```

**Comment by jacobp100** [comment]
It would throw at the RN point. Do you think we should check here?

**Comment by kristerkari** [comment]
I think that it's better to be consistent and always throw is something is not supported.

File: src/transforms/textDecorationLine.js (hunk 1 above)
Comment (jacobp100) [reply src/transforms/textDecorationLine.js]: Oh that's true. How would we solve that though. What if RN added a new value that was meant to be put after the current last value?

Comment (kristerkari) [reply src/transforms/textDecorationLine.js]: You could use the compare function from here to only reorder those two values but not touch the other ones:
https://github.com/styled-components/css-to-react-native/issues/64#issuecomment-363865691

Comment (kristerkari) [reply src/transforms/textDecorationLine.js]: at least that way it's somewhat future proof

**Comment by jacobp100** [comment]
What about when RN adds support for more properties? If we throw, there's zero change we'll have support for new features, whereas with a loose approach, it's probably 50/50 (like when percentages were added, we automatically supported them in a lot of places).

**Comment by kristerkari** [comment]
> What about when RN adds support for more properties?

Yeah, that's a good point. Maybe it's better to let RN throw.

File: src/transforms/textDecorationLine.js (hunk 1 above)
Comment (kristerkari) [reply src/transforms/textDecorationLine.js]: ping @jacobp100 do you think that this should be changed or left like it is now?

Comment (jacobp100) [reply src/transforms/textDecorationLine.js]: I'll probably leave as-is. Tbh I think they'd switch to an array of values if they added new values (like `fontVariant`)

**Pull request merged**